### PR TITLE
Acr tests cleanup

### PR DESCRIPTION
**Pull request by anamnavi**
# PR Summary
This PR cleans up modules and scripts that get published to ACR by deleting them after the tests have run.

Also make Install-PSResource case insensitve for ACR.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by alerickson — APPROVED** [review]

File: .ci/test.yml [hunk 1]
```diff
@@ -90,6 +90,25 @@ jobs:
...
+      inline: |
+        $acrRepositoryNamesFolder = Join-Path -Path ([Environment]::GetFolderPath([System.Environment+SpecialFolder]::LocalApplicationData)) -ChildPath 'TempModules'
+        if (Test-Path -Path $acrRepositoryNamesFolder) {
+          Write-Verbose -Verbose "Deleting existing temp acr repository names folder path: $acrRepositoryNamesFolder"
+          Remove-Item -Path $acrRepositoryNamesFolder -Recurse -Force -ErrorAction Ignore
+        }
+        if (! (Test-Path -Path $acrRepositoryNamesFolder)) {
+          Write-Verbose -Verbose "Creating new folder for acr repository names file to be placed with path: $acrRepositoryNamesFolder"
+          $null = New-Item -Path $acrRepositoryNamesFolder -ItemType Directory
+        }
```
Comment (adityapatwardhan) [inline .ci/test.yml]:
```suggestion
        Write-Verbose -Verbose "Creating new folder for acr repository names file to be placed with path: $acrRepositoryNamesFolder"
        $null = New-Item -Path $acrRepositoryNamesFolder -ItemType Directory
```

It seems that at this point, the folder will never be there. So no need to check.

File: src/code/ACRServerAPICalls.cs [hunk 2]
```diff
@@ -305,13 +305,14 @@ public override Stream InstallPackage(string packageName, string packageVersion,
         }
 
         private Stream InstallVersion(
-            string moduleName,
+            string packageName,
```
Comment (adityapatwardhan) [inline src/code/ACRServerAPICalls.cs]: This file is a product change. I think it should be a separate PR.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

**Review by adityapatwardhan — APPROVED** [review]
